### ISSUE TITLE
💅 Improve noSyncScripts rule

### ISSUE THREAD

**Issue by THernandez03** [biomejs/biome#8144]
### Environment information

```bash
» npx @biomejs/biome rage --linter
CLI:
  Version:                      2.3.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v25.2.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    nursery/noSyncScripts

Workspace:
  Open Documents:               0
```

### Rule name

noSyncScripts

### Playground link

https://github.com/THernandez03/biome-noSyncScripts

### Expected result

The `noSyncScripts` rule should be updated to:

1. **Skip validation** for `<script>` tags with `type="module"`
2. Only flag classic `<script>` tags without `defer` or `async` attributes
3. Optionally provide an informational message explaining that module scripts are already deferred

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Thank you @THernandez03 

A couple of things:
- for points 1 and 2, can you also share **why**? 
- can you update the reproduction by adding a valid file? The rule, as per documentation, is only available in JSX

**Comment by Netail** [comment]
`type="module"` indeed does not seem to block rendering; https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/script#blocking. The same effect as `defer`

> The rule, as per documentation, is only available in JSX

Also the html analyzer actually, but those don't seem to be displayed on the website yet.

**Comment by THernandez03** [comment]
Hey guys, first of all, thanks for the quick response and thanks for the awesome work you're doing 🙌.

I just updated the issue description for better understanding the points
